process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+            else:
+                url_FHPRS = ''
+                url_StG_FHPRS = ''
+            all_url =  url_RPKM + url_coverage + url_FHPRS + url_StG_RPKM + url_StG_coverage + url_StG_FHPRS
+            output = download_data(all_url)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def process_TSO(self):
+        """
+        Process TSO500 Runs
+        """
+        # destination for the files to be downloaded to:
+        destination_of_Results = config.TSO_destination_of_Results
+        # create folder woth project name
+        for file_path in self.project_files:
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            df_results = csv_file[csv_file['type'] == 'Results']
+            df_coverage = csv_file[csv_file['type'] == 'coverage']
+            df_sompy = csv_file[csv_file['type'] == 'sompy']
+            search_results1 = re.search(r"_(TSO\d+).csv", file_path)
+            search_results2 = re.search(r"002(_\S+)_TSO", file_path)
+            folder_name = (
+            search_results1.group(1) + search_results2.group(1) + "_AUTOMATE_DUTY_TEST"
+            )
+            path_to_folder = destination_of_Results + "\\" + folder_name
+            # create subfolders called coverage and Results and download data:
+            destination_of_Coverage = path_to_folder + "\\coverage"
+            destination_of_extracted_Results = path_to_folder + "\\Results"
+            destination_of_sompy = path_to_folder
+            os.mkdir(path_to_folder)
+            os.mkdir(destination_of_Coverage)
+            os.mkdir(destination_of_extracted_Results)
+            url_coverage = get_data(df_coverage, '"{}"'.format(destination_of_Coverage))
+            url_sompy = get_data(df_sompy, '"{}"'.format(destination_of_sompy))
+            url_results = get_data(df_results, '"{}"'.format(path_to_folder))
+            all_url = url_coverage + url_sompy + url_results
+            output = download_data(all_url)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def archive_files(self):
+        """
+        Modify the csv file to remove data from it and insert a message of when it was compelted and by whome.
+        """
+        # get the user infromation and current date
+        #user = os.getlogin() -> does not work when executing the script from PowerShell
+        cur_time = datetime.now()
+        cur_time_string = cur_time.strftime("%Y_%m_%d__%H_%M_%S_")
+        # message to be added to the csv file
+        message = "Processed_on_{}".format(
+            cur_time_string
+        )
+        # Repalce data in csv file with the message and move the files to archive folder.
+        for file_path in self.project_files:

```

Comment:
rather than doing them all at the end, could it be done when that file has finished downloading?

Reply:
not sure I understand what you mean. In the current setup the csv file is renames and moved to archive folder after all the files in the csv have been downloaded.

Reply:
I was thinking about scenarios where there are multiple fileruns to be processed, but I have missed that it's within a for loop